docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,595 @@
+                item_id=item_id, item_name=item_name, quantity=quantity
+            )
+
+    return {"item": grocery_list[item_id]}
+    ```
+
+    You might notice Pylance adds inlay hints with the function return type, as well as the types for `item_ids` and `item_id`. You can double click on each suggestion to insert them into the code:
+    ![Inlay function return and variable type hints being displayed by Pylance throughout the sample code.](images/fastapi-tutorial/pylance_inlay_hints.png)
+
+Now let's see if this route is working as expected. The fastest way to do so is to leverage both VS Code's debugger as well as FastAPI's `/docs` endpoint, which provides information about all the available API routes and allows you to interact with the API to explore their parameters and responses. This documentation is generated dynamically based on the metadata and type hints defined in the FastAPI application.
+
+4. Add a breakpoint next to the if quantity <= 0 statement, by clicking on the left margin of the line number (or `kb(editor.debug.action.toggleBreakpoint)`). This will make the debugger stop prior to the execution of that line, allowing us to inspect the code line by line.
+    ![Breakpoint set next to the first line in the add_item function.](images/fastapi-tutorial/debugger_breakpoint.png)
+
+5. Start the debugger (`kb(workbench.action.debug.start)`), and then navigate to http://127.0.0.1:8000/docs in the browser.
+
+    You'll see a Swagger interface with the two endpoints available in the app: `/items` and root (`/`).
+    ![Swagger UI displaying two endpoints: /items and /](images/fastapi-tutorial/fastapi_first_swagger_page.png)
+
+6. Click on the down arrow next to the `/items` route to expand it, and click on the "Try it out" button that appears on the right side.
+    ![Try it out button displayed next to the /items route in the Swagger UI.](images/fastapi-tutorial/fastapi_tryitout_button.png)
+
+7. Add a grocery list item by passing a string to the "item" field, and a number to "quantity". For example, you could provide `{"item" : "apple", "quantity": 2 }`.
+
+8. Click on the "Execute" button.
+    ![Execute button displayed below the /items route.](images/fastapi-tutorial/fastapi_execute_button.png)
+
+9. Open VS Code again and notice the debugger has stopped at the breakpoint we set earlier.
+    ![Debugger stopped at the breakpoint set in the add_item function.](images/fastapi-tutorial/fastapi_breakpoint_hit.png)
+
+On the left side, you will see all local and global variables defined at this point in the Variables window, under the **Run and Debug** view:
+
+![Variables window displayed in the Run and Debug view, with the item and grocery_list variables highlighted.](images/fastapi-tutorial/fastapi_debugger_variables.png)
+
+Now let’s leverage VS Code's Debug Console to do some exploration.
+
+10.	Select the `quantity <= 0` statement, right-click on the editor and select **Evaluate in Debug Console**:
```

Comment:
Would it be better to say "highlight" here? I could see users potentially thinking the line being highlighted by the breakpoint is sufficient for this step since the screenshot doesn't show just `quantity <= 0` being highlighted.

Reply:
I ended up changing the screenshot to show the selection (mostly because I think highlighting is more commonly used in the context of symbols instead of expressions)